Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -318,51 +356,4 @@ - (SentrySpanStatus)spanStatusForHttpResponseStatusCode:(NSInteger)statusCode
     return kSentrySpanStatusUndefined;
 }
 
-- (NSString *)removeSentryKeysFromBaggage:(NSString *)baggage
```

Comment:
`m`: Why can we ditch that code?

Reply:
I thought that we wouldn't run into the same problem with the wrong headers anymore but looking at it again, we can still run into the problem where NSURLSession itself has cached the old headers. Restoring this functionality, will add tests after lunch (I need to run an errand now).

Reply:
Now we are injecting the header directly into NSURLRequest, the session will not cache this.

Reply:
Right, that was also my first instinct but Philip's questions made me question myself 😅

I've removed it yet again and tested it, and indeed we don't need this workaround anymore.